src/components/BlogSearchForm/BlogSearchForm.jsx

```diff
@@ -1,70 +1,194 @@
 import React from "react";
 import { navigate } from "gatsby";
-import { Button, Col, Form, FormGroup, Input, Label, Row } from "reactstrap";
+import { Col, Form, FormGroup, Input, Label, Button } from "reactstrap";
+import { IconContext } from "react-icons";
+import { FaSearch } from "react-icons/fa";
+import blogList from "../../generated/blog-result.json";
+const moment = require("moment");
+moment.locale("en");
 
-const SearchForm = ({ query, filter }) => (
-  <Form role="search" method="GET">
-    <Row form id="search-form">
-      <Col md="2">
-        <FormGroup>
-          <Label for="searchFilter">
-            <h6>Filter</h6>
-          </Label>
-          <Input
-            type="select"
-            name="filter"
-            id="search-tag"
-            bsSize="lg"
-            aria-controls="search-results-count"
-            onChange={e =>
-              navigate(
-                `${
-                  location.pathname
-                }?keywords=${query}&filter=${encodeURIComponent(
-                  e.target.value
-                )}`
-              )
-            }
-            value={filter}
-          >
-            <option value="">All</option>
-            <option value="gsoc">Google Summer Of Code</option>
-            <option value="project">Project</option>
-            <option value="TeraSaturday">TeraSaturday</option>
-            <option value="update">Update</option>
-          </Input>
-        </FormGroup>
-      </Col>
-      <Col md="8">
-        <FormGroup>
-          <Label for="searchQuery">
-            <h6>Search</h6>
-          </Label>
-          <Input
-            type="search"
-            id="search-input"
-            name="keywords"
-            bsSize="lg"
-            aria-controls="search-results-count"
-            onChange={e =>
-              navigate(
-                `${location.pathname}?keywords=${encodeURIComponent(
-                  e.target.value
-                )}&filter=${filter}`
-              )
-            }
-            placeholder="Search..."
-            value={query}
-          />
-        </FormGroup>
-      </Col>
-      <Col md="2">
-        <Button type="submit" color="primary" size="lg" id="search-btn">
-          Search
-        </Button>
-      </Col>
-    </Row>
-  </Form>
-);
+const SearchForm = ({ query, tag, author, ddate }, props) => {
+  let srcLocation = props.location;
+
+  if (typeof window !== `undefined`) {
+    srcLocation = location.search;
+  }
+  let urlTag = new URLSearchParams(srcLocation).get("tag") || "";
+  let urlAuthor = new URLSearchParams(srcLocation).get("author") || "";
+  let urlDdate = new URLSearchParams(srcLocation).get("ddate") || "";
+  let authorList = [];
+  let tagList = [];
+
+  blogList.forEach((blog) => {
+    authorList.push(blog.author);
+    blog.tags.forEach((tag) => {
+      tagList.push(tag);
+    });
+  });
+
+  authorList = [...new Set(authorList)];
+  tagList = [...new Set(tagList)];
```

Comment:
You might as well start with a set of your going to do this.

Reply:
Map requires to array , Set do not work with array, that's why i took a array and then removed duplicate entry so that i can  dispaly it

Reply:
```
s = new Set()
s.add("test")
s.add("test1")
s.add("test")
[...s.values()].map(k => k)
```

Reply:
done